neurons/validator.py

```diff
@@ -519,6 +440,132 @@ def update_models(self):
 
         bt.logging.info("Exiting update models loop.")
 
+    def _should_retry_model(self, epsilon_func, curr_block, eval_history) -> bool:
+        # If the model has never been evaluated, we should retry it.
+        if not eval_history:
+            return True
+
+        # Find the most recent successful eval.
+        last_successful_eval = None
+        for eval_result in reversed(eval_history):
+            if eval_result.avg_loss != math.inf:
+                last_successful_eval = eval_result
+                break
+
+        if last_successful_eval:
+            # If this model had worse loss than the top model during the last eval, no need to retry.
+            if last_successful_eval.score > last_successful_eval.winning_model_score:
```

Comment:
This rename is a little awkward since we are saying if the 'score' is higher to not retry.

Reply:
Ack. Added a comment to help make this clearer